streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -749,9 +749,11 @@ void runOnce() {
             // multiple iterations with reasonably large max.num.records and hence is less vulnerable to outliers
             taskManager.recordTaskProcessRatio(totalProcessLatency, now);
 
-            log.info("Processed {} total records, ran {} punctuators, and committed {} total tasks " +
-                        "for active tasks {} and standby tasks {}",
-                     totalProcessed, totalPunctuated, totalCommitted, taskManager.activeTaskIds(), taskManager.standbyTaskIds());
+            // Don't log summary if no new records were processed to avoid spamming logs for low-traffic topics
+            if (totalProcessed > 0 || totalPunctuated > 0 || totalCommitted > 0) {
+                log.info("Processed {} total records, ran {} punctuators, and committed {} total tasks",
```

Comment:
Maybe we do this else where but can we log at DEBUG if there is nothing done and there is tasks assigned to the thread? It might be good to be able to confirm that its not getting stuck here or its actually not polling anything.

Reply:
I have a similar question here, other than that LGTM

Reply:
Good thought, but don't worry, we already do. John added some pretty thorough logging at the DEBUG level after an escalation a while back